rfd/0174-kube-rbac-bootstrap.md

```diff
@@ -0,0 +1,158 @@
+---
+authors: Anton Miniailo (anton@goteleport.com)
+state: draft
+---
+
+# RFD 174 - Kubernetes RBAC Bootstrap
+
+## Required Approvers
+
+- Engineering: `@rosstimothy` && (`@tigrato` || `@hugoShaka`)
+- Product: `@klizhentas` || `@xinding33`
+
+## What
+
+Improve RBAC setup capabilities for Kubernetes and make it easier to use with Teleport by exposing
+the default Kubernetes user-facing roles and adding the ability to define RBAC setup to be
+automatically provisioned to Kubernetes clusters.
+
+## Why
+
+Currently, users have to rely on third-party tools or manual actions
+when making sure their Kubernetes clusters have the correct roles/role bindings and that those are in sync with the RBAC setup defined in Teleport.
+The changes we propose will make it easier for clients to start using Teleport for accessing Kubernetes clusters.
+
+## Scope
+
+This RFD is focused only on Kubernetes resources related to RBAC functionality, other types of resources are out of scope,
+though they might be added in the future.
+
+## Details
+
+### Exposing the default Kubernetes user facing cluster roles (cluster-admin, view, edit)
+
+Kubernetes clusters have default user-facing cluster roles: cluster-admin, view, edit. By default they are not usable, because
+they are not exposed through role bindings to any subjects (the cluster-admin role has internal Kubernetes bindings). 
+Currently, we only expose the cluster-admin role on GKE
+installations, since system:masters group is not available for impersonation there. We will expand on this and when
+discovering a cluster/installing kube-agent/creating a service account, we will
+create cluster role bindings for those roles, accordingly linking them to the groups "default-cluster-admin", "default-view" and
```

Comment:
Yep, I've described this option in the alternative. But I think overall it's better to go with the "expose by default" route, since combined with our now available fine-tuning of kubernetes resources access in Teleport roles it can make it easier for the new users and to those users who don't need complicated RBAC setup. We could also depend on that default in our documentation and simplify those as well (which we can't really do if we rely on kube provision path). But ultimately it's a product decision.